search/node-search.go

```diff
@@ -58,11 +83,16 @@ func MapKeyValuePairs(content []*yaml.Node) map[string]keyValue {
 
 // Helper in which path aligns with matching key node and needs
 // further to determine further traversal
-func resolveMatchingPath(key *yaml.Node, val *yaml.Node, path ...string) (*yaml.Node, error) {
+func resolveMatchingPath(key *yaml.Node, val *yaml.Node, seeking Seeking, path ...string) (*yaml.Node, error) {
 	if len(path) == 1 {
-		return key, nil
+		switch seeking {
+		case VALUE:
+			return val, nil
+		default:
```

Comment:
Not sure what is standard in Go, but should this be the case for KEY? Do we need a default in Go? or should default return an error?

Reply:
got rid of seeking